File: src/main/java/me/viniciusarnhold/altaria/apis/HttpManager.java
package me.viniciusarnhold.altaria.apis;

import okhttp3.OkHttpClient;
import org.jetbrains.annotations.NotNull;


/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
public class HttpManager {

    @NotNull
    private static final OkHttpClient defaultHttpClient;

    @NotNull
    private static HttpManager ourInstance = new HttpManager();

    static {
        defaultHttpClient = new OkHttpClient.Builder().build();
    }

    private HttpManager() {
    }


    @NotNull
    public static HttpManager getInstance() {
        return ourInstance;
    }


    @NotNull
    public OkHttpClient getDefaultClient() {
        return defaultHttpClient;
    }
}

File: core/src/main/kotlin/me/viniciusarnhold/altaria/apis/HttpManager.kt
package me.viniciusarnhold.altaria.apis

import okhttp3.OkHttpClient


/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
class HttpManager private constructor() {


    val defaultClient: OkHttpClient
        get() = defaultHttpClient

    companion object {

        private val defaultHttpClient: OkHttpClient

        val instance = HttpManager()

        init {
            defaultHttpClient = OkHttpClient.Builder().build()
        }
    }
}
